### PR TITLE
[IMP] l10n_mx: Import taxes are supported now

### PR DESCRIPTION
**Pull request by luistorresm**
User Case:
The Company A buy some products from a Foreign Customer, and when they
are received in Mexico, the Broker Partner send an invoice to Company A
by the importation taxes.

In this case, when Company A, make the payment to the broker, the taxes
could be accredited for the company and in your case must be reported in
the DIOT report.

More info about the configuration in the README file.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by luistorresm** [comment]
Hello @nhomar @moylop260 

Could you approve this PR, please?

@hbto was approved.

Regards

File: addons/l10n_mx/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                l.l10n_mx_invoice_broker_id):
+            if line.l10n_mx_invoice_broker_id == self:
+                self.message_post(body=_(
+                    'The Overseas Invoice cannot be the same that the'
+                    'related invoice'))
+                continue
+            tax_ids = self.env['account.tax']
+            for tax in line.tax_ids:
+                tax_ids |= tax
+                for child in tax.children_tax_ids:
```
Comment (moylop260) [inline addons/l10n_mx/models/account_move.py]: what about using filtered?

File: addons/l10n_mx/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                'name': (line.name or '').split('\n')[0][:64],
+                'move_id': line.move_id,
+                'company_id': line.move_id.company_id.id,
+                'account_id': line.account_id.id,
+                'product_id': line.product_id.id,
+                'tax_ids': [(6, 0, tax_ids.ids)],
+                'partner_id': partner_id,
+                'exclude_from_invoice_tab': True,
+                'quantity': 1,
+                'price_unit': price if price > 0 else -price,
```
Comment (moylop260) [inline addons/l10n_mx/models/account_move.py]: What about using “abs(price)”?

Comment (luistorresm) [reply addons/l10n_mx/models/account_move.py]: In some cases must be negative

Comment (moylop260) [reply addons/l10n_mx/models/account_move.py]: What cases?

```python
if price >0:
  # it is positive
  price
else:
  # it is negative
  -price # now it is positive
```

What am I missing?

**Comment by luistorresm** [comment]
Done @moylop260 

Regards

File: addons/l10n_mx/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        lines_2_create = []
+        for line in self.line_ids.filtered(
+                lambda l: not l.quantity and l.tax_ids and
+                l.l10n_mx_invoice_broker_id):
+            if line.l10n_mx_invoice_broker_id == self:
+                self.message_post(body=_(
+                    'The Overseas Invoice cannot be the same that the'
+                    'related invoice'))
+                continue
+            tax_ids = self.env['account.tax']
```
Comment (moylop260) [inline addons/l10n_mx/models/account_move.py]: What about using the following way?
`tax_ids = line.tax_ids | line.tax_ids.mapped('children_tax_ids').filtered(lambda child: child.type_tax_use != 'none')`

File: addons/l10n_mx/models/account_move.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                'price_subtotal': abs(price),
+            }
+            aml_obj.new(move_line_dict)
+
+            price = -price
+
+            ml_dict = dict(
+                move_line_dict,
+                tax_ids=[],
+                price_unit=price if price < 0 else -price,
```
Comment (moylop260) [inline addons/l10n_mx/models/account_move.py]: Is it not the same case of `abs`?

This case is `-abs(price)`

**Comment by luistorresm** [comment]
Done @moylop260 

Thanks

**Comment by jpp-odoo** [comment]
@jco-odoo Could you review this PR ? Thanks.

**Comment by jco-odoo** [comment]
@luistorresm @nhomar @moylop260 I think you can guess that this is difficult to digest on our side: 

we prefer to avoid:
- a python tax
- magically creating lines within the invoice (it does not mean we can not create it somewhere else e.g.)
- linking account move lines with partners

However, there is probably a need with the imports and the question arises of course if this is only related to Mexico or if other countries have the same problems.  @bfr-o can check further.

File: addons/l10n_mx/models/account_move.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        res = super(
+            AccountMove, self)._move_autocomplete_invoice_lines_values()
+        return res
+
+    def _l10n_mx_get_import_tax_base_amount_lines(self):
+        """Appends two lines for each one of the lines where the quantity is
+        zero, as the value of the lines added are need in cash basis taxes to
+        properly compute the taxes paid"""
+        self.ensure_one()
+        foreign = self.line_ids.mapped('l10n_mx_invoice_broker_id.partner_id')
```
Comment (nhomar) [inline addons/l10n_mx/models/account_move.py:21]: commercial_entitty_id

**Comment by william-andre** [comment]
I think this PR is a bit outdated. Is it still needed?
Should we close it?

**Pull request closed without merging**